src/js/background.js

```diff
@@ -93,6 +93,19 @@ function Badger() {
     await dntHashesPromise;
     await tabDataPromise;
 
+    // gather list of url schemes that firstparty content scripts run on
+    let manifestBlob = await chrome.runtime.getManifest();
+    let firstParties = [];
+
+    for (let contentScriptObj of manifestBlob.content_scripts) {
+      // only include parts from content scripts that have firstparties entries
+      if (contentScriptObj.js[0].includes("firstparties")) {
+        firstParties.push(contentScriptObj.matches);
+      }
+    }
+    // set list of firstparty url schemes onto badger object
+    self.firstPartiesList = [].concat.apply([], firstParties);
```

Comment:
Since the `firstPartyProtectionsEnabled()` method got moved to utils and now only the `badger.firstPartiesList` gets set on startup, I think that resolves this comment, yes?

Reply:
Well, do we need to collect match patterns of first party protection scripts on Badger startup? If we don't, let's have `badger.firstPartiesList` initialization happen the first time we need it. My code sample above illustrates an approach that accomplishes this.

Reply:
I didn't follow the exact pattern above, but stuck to the same idea on fetching the firstparties list only the first time its needed rather than during startup. Thank you!

Reply:
We don't need `badger.firstPartiesList` though anywhere else though, and we shouldn't have to pass in the `badger` object with every call to this function either. Keeping local state is what the `firstPartiesList` local-to-the-function-[closure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures) variable in the code snippet above accomplishes.

Reply:
Oh, right! Good call, then. I've changed the method to that, then.